go/vt/vtgateproxy/discovery.go

```diff
@@ -319,11 +320,16 @@ func (b *JSONGateResolverBuilder) parse() (bool, error) {
 	// target lock and then comparing the previous targets with the current
 	// targets and only resetting pools which disappear.
 	targetCount.ResetAll()
-
+	idx := getClientIndex()
 	for poolType := range targets {
 		b.sorter.shuffleSort(targets[poolType], b.affinityField, b.affinityValue)
```

Comment:
Presumably for this approach we'd want to _not_ do the random shuffleSort and instead make sure that the `targets` list is in a stable order?